modules/gltf/gltf_document.cpp

```diff
@@ -3096,7 +3096,7 @@ Error GLTFDocument::_parse_images(Ref<GLTFState> p_state, const String &p_base_p
 		if (d.has("uri")) {
 			// Handles the first two bullet points from the spec (embedded data, or external file).
 			String uri = d["uri"];
-			image_name = uri;
```

Comment:
we should try and use the d["name"] if it exists. otherwise, fallback to textures or material names
NOTE: we should check some files from a few different DCC to see how they handle this.

And then fallback to index only if no name

images->name exists in blender embedded exports.

Reply:
Updated. For now I'm only using image["name"]. using material name is rather tricky, especially if they use multiple texture maps which all need unique names assigned.